### PR TITLE
[Network] `az network lb create`: Support cross-subscription resource ID

### LINKED ISSUE #28003: Support cross-subscription resource ID
### Preconditions

- [X] No need to upgrade Python SDK or the Python SDK is ready.

### Related command

**az network lb create**
az network lb frontend-ip create
az network lb frontend-ip update
az network lb address-pool create 
az network lb address-pool update
az network lb address-pool address add
az network lb address-pool address update

### Resource Provider

Microsoft.Network/loadBalancers

### Description of Feature or Work Requested

The SLB team is launching a cross-subscription feature for Azure Load Balancer. Currently, the CLI parameters (vnet, subnet, public IP addresses) accept resource IDs, but the IDs are limited to same subscription as the LB. 

CLI should now support accepting resource IDs without this restriction and accepts Ids across subscriptions. 

In addition, the command "az network lb create" vnet parameter should be able to take in an ID value (currently only accepts a name). 

### Minimum API Version Required

2023-04-01

### Swagger PR link / SDK link

 N/A

### Request Example

_No response_

### Target Date

2024-02-02

### PM Contact

mahipdeora, anniefang

### Engineer Contact

irrogozh

### Additional context

Please reach out on teams for additional questions

### PR DESCRIPTION
**Pull request by ReaNAiveD**
**Related command**

* `az network lb create`

**Description**

Resolve #28003 
Support cross subscription resource id for `--subnet` and `--public-ip-address`

**Testing Guide**

`network lb create -g myRG -n myLB --subnet SubnetId --sku Standard`
`network lb create -g myRG -n myLB --public-ip-address PublicIpAddressId --sku Standard`

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli-core/azure/cli/core/commands/arm.py [hunk 1]
```diff
@@ -175,11 +175,14 @@ def __new__(cls, val):
         return str.__new__(cls, val)
 
 
-def resource_exists(cli_ctx, resource_group, name, namespace, type, **_):  # pylint: disable=redefined-builtin
+def resource_exists(cli_ctx, subscription, resource_group, name, namespace, type,
+                    **_):  # pylint: disable=redefined-builtin
```
Comment (zhoxing-ms) [inline src/azure-cli-core/azure/cli/core/commands/arm.py:179]: Have you confirmed that inserting `subscription` in the middle of the parameter list will not cause parameter misalignment issues in previous calls?

Comment (zhoxing-ms) [reply src/azure-cli-core/azure/cli/core/commands/arm.py:179]: Please note that this is a method in the azure-cli-core, so we also need to consider the invocation of other external dependencies such as CLI extensions. If you are not 100% confident, it is recommended to insert `subscription` parameter at the end of the parameter list.

**Review by zhoxing-ms — APPROVED** [review]

File: src/azure-cli-core/azure/cli/core/commands/arm.py (hunk 1 above)
Comment (ReaNAiveD) [reply src/azure-cli-core/azure/cli/core/commands/arm.py:179]: Yes, only two functions(`get_folded_parameter_validator(core/commands/templated_create.py)` and `_resource_not_exists(vm/_actions.py)`) in azure-cli and azure-cli-extensions use this function and I have update the call in `_resource_not_exists` correspondingly.

**Review by necusjz — APPROVED** [review]

**Review by jsntcy — APPROVED** [review]

**Pull request merged**